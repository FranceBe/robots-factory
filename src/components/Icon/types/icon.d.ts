import { FontAwesomeIconProps } from '@fortawesome/react-fontawesome'

export type IconProps = Partial<FontAwesomeIconProps> & {
  type: IconType
}

export type IconType = 'robot' | 'spinner' | 'failure' | 'success' | 'moving'
