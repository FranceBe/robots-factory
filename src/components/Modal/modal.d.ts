import { ReactElement } from 'react'

export type ModalProps = {
  isOpen: boolean
  content: string | ReactElement
  buttonText: string
  onButtonClick: () => void
}
