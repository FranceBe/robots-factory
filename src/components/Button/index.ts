import ButtonDefault, { Button } from 'components/Button/Button.component'

export { Button }

export default ButtonDefault
