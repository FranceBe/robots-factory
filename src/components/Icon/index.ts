import { FAIconByIconProps, Icon } from 'components/Icon/Icon.component'

export { FAIconByIconProps, Icon }
