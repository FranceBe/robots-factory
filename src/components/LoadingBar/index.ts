import { LoadingBar } from 'components/LoadingBar/LoadingBar.component'

export { LoadingBar }
