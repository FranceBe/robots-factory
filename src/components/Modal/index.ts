import ModalDefault, { Modal } from 'components/Modal/Modal.component'

export { Modal }

export default ModalDefault
