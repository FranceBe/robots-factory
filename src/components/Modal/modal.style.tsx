import styled from 'styled-components'
import { fontSizes, palette } from 'styles/variables'

export const ModalContent = styled.div`
  text-align: center;
  font-size: ${fontSizes.medium};
  span {
    color: ${palette.green_primary};
  }
`

export const modalStyle = {
  content: {
    alignItems: 'center',
    display: 'flex',
    justifyContent: 'center',
    left: '50%',
    marginRight: '-50%',
    right: '30%',
    top: '50%',
    transform: 'translate(-50%, -50%)',
  },
  overlay: {
    backgroundColor: 'rgba(0, 0, 0, 0.75)',
    zIndex: 4,
  },
}
