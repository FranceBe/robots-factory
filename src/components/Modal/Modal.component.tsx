// Modal component
// I used react-modal to create a dynamic modal
import { Button } from 'components/Button'
import { ModalProps } from 'components/Modal/modal'
import { ModalContent, modalStyle } from 'components/Modal/modal.style'
import React from 'react'
import ReactModal from 'react-modal'
import { ThemeType } from 'utils/common.enum'

export const Modal: React.FC<ModalProps> = ({ isOpen, children, buttonText, onButtonClick }) => {
  return (
    <ReactModal isOpen={isOpen} ariaHideApp={false} style={modalStyle}>
      <ModalContent>
        <div>{children}</div>
        <br />
        <Button buttonType={ThemeType.secondary} onClick={onButtonClick}>
          {buttonText}
        </Button>
      </ModalContent>
    </ReactModal>
  )
}

export default Modal
