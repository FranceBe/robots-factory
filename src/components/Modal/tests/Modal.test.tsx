import '@testing-library/jest-dom/extend-expect'

import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Modal } from 'components/Modal'
import { ModalProps } from 'components/Modal/types/modal'
import React from 'react'

describe('Modal component', () => {
  const onClickMock = jest.fn()
  const initTest = (props?: Partial<ModalProps>) =>
    render(
      <Modal
        isOpen={true}
        content={<p>Modal Content</p>}
        buttonText={'Click'}
        onButtonClick={onClickMock}
        {...props}
      />,
    )
  describe('Given modal is open', () => {
    it('should match snapshot', () => {
      const { container } = initTest()

      expect(container.firstChild).toMatchSnapshot()
    })
    it('should display content', () => {
      initTest()

      expect(screen.getByText('Modal Content')).toBeInTheDocument()
    })
    it('should display a button', () => {
      initTest()

      expect(screen.getByRole('button')).toBeInTheDocument()
      expect(screen.getByText('Click')).toBeInTheDocument()
    })
    it('should call onClickMock when button is clicked', () => {
      initTest()

      const button = screen.getByRole('button')
      userEvent.click(button)

      expect(onClickMock).toHaveBeenCalledTimes(1)
    })
  })
  describe('Given modal is closed', () => {
    it('should not display content nor button', () => {
      initTest({ isOpen: false })

      expect(screen.queryByText('Modal Content')).not.toBeInTheDocument()
      expect(screen.queryByRole('button')).not.toBeInTheDocument()
    })
  })
})
