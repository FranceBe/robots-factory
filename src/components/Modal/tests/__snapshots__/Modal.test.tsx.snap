// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component Given modal is open should match snapshot 1`] = `null`;
